set.mm

```diff
@@ -41727,6 +41728,13 @@ choice between (what we call) a "definitional form" where the shorter
        4-Jan-2017.) $)
     rspcedv $p |- ( ph -> ( ch -> E. x e. B ps ) ) $=
       ( cv wceq wa biimprd rspcimedv ) ABCDEFGADIEJKBCHLM $.
+
+    rspcebdv.1 $e |- ( ( ph /\ ps ) -> x = A ) $.
+    $( Restricted existential specialization, using implicit substitution in
+       both directions.  (Contributed by AV, 8-Jan-2022.) $)
+    rspcebdv $p |- ( ph -> ( E. x e. B ps <-> ch ) ) $=
+      ( wrex wi wa cv wceq wb syldan biimpd expcom pm2.43b rexlimdvw rspcedv
```

Comment:
Are you sure this is a worthwhile result ? Keeping ph implicit, the hypotheses read:
A e. B & ( ps <-> ( x = A /\ ch ) )
and since DV(x,ch), the result follows.  That might be your proof, I haven't checked.  Does stating the hypotheses as in your version bring something more than with the hypotheses stated as here ?

Reply:
this theorem is just a combination of ~rspcedv and ~rexlimdvw, each of them proving one direction of the biconditional. The main (small) advantage (and interesting part) is that the hypothesis rspcdv.2 $e |- ( ( ph /\ x = A ) -> ( ps <-> ch ) ) is used for both directions. And this theorem provides an equivalence of a restricted existential quantification without using another (restricted) existential quantification. Finally, this theorem is used to prove ~fusgr2wsp2nb, shortening its proof a little bit.

Reply:
Can you try using scripts/min.cmd to "minimize_with rspcebdv" the output of "show usage rspce*,rexlim*" (or some other  set that you see fit) ? (generally, I remove from the list of theorems to be minimized fourier*, stowei*, stirlin* and other huge proofs; you can also remove all proofs in mathboxes)

Note that I'm not opposing moving that theorem to Main.

Reply:
I ran the script with "show usage rspcedv*,rexlimdv*"(there were too many proofs to be checked with your pattern), which checked about 1500 proofs. But no proof could have been shortened. Maybe there is really no other use case for this theorem (yet), or it is too complicated to be found by automatic minimzation.